### ISSUE TITLE
QuarkusGenerator should extend from JavaExecGenerator - Refactor

### ISSUE THREAD

**Issue by manusa** [eclipse/jkube#430]
### Description

[QuarkusGenerator](https://github.com/eclipse/jkube/blob/55489344dd0395b5f5c4fc72aa5858ca2aff6f01/jkube-kit/jkube-kit-quarkus/src/main/java/org/eclipse/jkube/quarkus/generator/QuarkusGenerator.java#L38) should extend from [JavaExecGenerator](https://github.com/eclipse/jkube/blob/092e24b3f6d41f40f0067350ca4365d9f4534419/jkube-kit/generator/java-exec/src/main/java/org/eclipse/jkube/generator/javaexec/JavaExecGenerator.java#L45) to take advantage of JavaExecGenerator defaults.
This should enable:
- Prometheus integration (expose 9778 port)
- `jkube.generator.fromMode` + Base image configuration (We need to adapt QuarkusGenerator own implementations to use the more abstract ones)

It's important to note that QuarkusGenerator behaves differently depending on the target (JVM or native), we need to consider these adaptations when using abstract defaults in JavaExecGenerator.

**Comment by rohanKanojia** [comment]
Is this still an issue? I can see `QuarkusGenerator` is extending `JavaExecGenerator` now:

https://github.com/eclipse/jkube/blob/master/jkube-kit/jkube-kit-quarkus/src/main/java/org/eclipse/jkube/quarkus/generator/QuarkusGenerator.java#L38

**Comment by manusa** [comment]
There are still thing pending on the Generator, but I think they can be covered on the scope of #450

**Comment by manusa** [comment]
Superseded by #473
